src/analysis/complex/circle.lean

```diff
@@ -93,8 +94,100 @@ rfl
 
 /-- The map `λ t, exp (t * I)` from `ℝ` to the unit circle in `ℂ`, considered as a homomorphism of
 groups. -/
-def exp_map_circle_hom : ℝ →+ (additive circle) :=
+@[simps] def exp_map_circle_hom : ℝ →+ (additive circle) :=
 { to_fun := exp_map_circle,
   map_zero' := by { rw exp_map_circle, convert of_mul_one, simp },
   map_add' := λ x y, show exp_map_circle (x + y) = (exp_map_circle x) * (exp_map_circle y),
     from subtype.ext $ by simp [exp_map_circle, exp_add, add_mul] }
+
+lemma exp_map_circle_hom_ker : exp_map_circle_hom.ker = add_subgroup.gmultiples (2 * π) :=
+begin
+  ext k,
+  have a0 : (0 : additive circle) = (1 : circle) := rfl,
+  have iI : ∀ (x y : ℂ), x * I = y * I ↔ x = y :=
+    λ _ _, ⟨mul_right_cancel' I_ne_zero, λ h, congr_arg _ h⟩,
+  suffices : (∃ (n : ℤ), (k : ℂ) = n * 2 * π) ↔ ∃ (n : ℤ), (n : ℝ) * 2 * π = k,
+  { simpa [add_subgroup.mem_gmultiples_iff, add_monoid_hom.mem_ker, subtype.ext_iff,
+           a0, exp_eq_one_iff, ←mul_assoc, iI] },
+  simp_rw @eq_comm _ _ k,
+  norm_cast,
+end
+
+lemma exp_map_circle_eq_one_iff {x : ℝ} :
+  exp_map_circle x = 1 ↔ x ∈ add_subgroup.gmultiples (2 * π) :=
+begin
+  simp [add_subgroup.mem_gmultiples_iff],
```

Comment:
Using `simp` in the middle of a proof rather than to finish it off is fragile, so use `squeeze_simp` to get:
```suggestion
  simp only [add_subgroup.mem_gmultiples_iff, gsmul_eq_mul],
```